### ISSUE TITLE
Docker Integration

### ISSUE THREAD

**Issue by josephcrawfordSRH** [philstainer/git-worktree#12]
I wanted to move over to the git-worktree workflow but we use docker at our company.  Does this plugin give a way to symlink a certain directory based on which worktree you are actively using?

We currently have our code in a structure like `/projects/project-1/.git`, `/projects/project-2/.git`, etc we map our docker containers to these directories so that the code is served from there.  However I was wondering if the git-worktree plugin had the capabilities on switching worktrees to change this directory, for instance if I were to clone to `/projects/src/project-1` could this project sym link the `/projects/project-1` directory to be /projects/src/project-1/work-tree-in-use/`?

**Comment by philstainer** [comment]
Short answer is no, not currently in its current state

I did want to add a feature that, on new worktree creation, you could run a script from your package.json/custom config. My use case was installing node_modules automatically etc...

In the end I decided not to add it as I wasn't sure there was a need for it and also is this the right place for this to exist? I'm not sure

For now, could you create a custom script for sym linking your directories?

**Comment by josephcrawfordSRH** [comment]
Yes I could create a custom script but that would be a pain having to execute it with every work tree switch as I am between projects many times a day.  I guess I have to say with my current setup until something is integrated for use with docker which stinks because using work trees looks so nice.

**Comment by josephcrawfordSRH** [comment]
@philstainer that would work perfectly, so long as the script can receive the path of the worktree I could do the symlink rather easily but without knowing the work tree path that would be difficult to do.

**Comment by josephcrawfordSRH** [comment]
@philstainer Have you looked at this and found it to not be as easy as initially thought?  I would love to get something in place to take advantage of work trees in the new year sometime :). 

I hope you had a happy and safe holiday(s).

**Comment by philstainer** [comment]
Thank you! I hope you had a nice break too 😃 

I did have a look into it and was a little tricky! I have been pretty busy with work and the holidays stuff. Hopefully have some time next week to play again

**Comment by josephcrawfordSRH** [comment]
@philstainer no problem at all I know things slow over the holiday's they sure did for me :). I just wanted to touch base and see if it was looked at or if the holidays made you forget about it.

**Comment by josephcrawfordSRH** [comment]
@philstainer thanks for digging into this, once figured out it will help tremendously.  I wonder if vscode is caching something which is causing the events to not trigger 🤷  I am not familiar with how vscode plugins work at the moment.
